ee/server/services/ddp-streamer/Server.ts

```diff
@@ -34,24 +38,32 @@ export class Server extends EventEmitter {
 
 	async call(client: Client, packet: IPacket): Promise<void> {
 		try {
+			// if method was not defined on DDP Streamer we fall back to Meteor
 			if (!this._methods.has(packet.method)) {
 				const result = await MeteorService.callMethodWithToken(client.userId, client.userToken, packet.method, packet.params);
 				if (result?.result) {
 					return this.result(client, packet, result.result);
 				}
 
-				throw new Error(`Method '${packet.method}' doesn't exist`);
+				throw new MeteorError(404, `Method '${packet.method}' not found`);
 			}
-			const fn = this._methods.get(packet.method);
 
+			const fn = this._methods.get(packet.method);
 			if (!fn) {
-				throw Error('method not found');
+				throw new MeteorError(404, `Method '${packet.method}' not found`);
 			}
 
 			const result = await fn.apply(client, packet.params);
 			return this.result(client, packet, result);
-		} catch (error) {
-			return this.result(client, packet, null, error.toString());
+		} catch (err: unknown) {
+			if (err instanceof MeteorError) {
+				return this.result(client, packet, null, err);
+			}
+
+			// default errors are logged to the console and redacted from the client
+			logger.error({ msg: 'Method call error', err });
+
+			return this.result(client, packet, null, new MeteorError(500, 'Internal server error'));
```

Comment:
I think this check should be inside the result function (or in a "send" function called by it), checking if the error is client save to return it or to print and return 500

Reply:
is your goal to remove the duplicated error handling? if that is the case, I can create a helper to do that, so the result would be something like:

```ts
catch (err: unknown) {
  return this.result(client, packet, null, handleInternalException(err));
}
// ...
catch (err: unknown) {
  return this.nosub(client, packet, handleInternalException(err));
}
```

so the logic to either log the error or return would be only into `handleInternalException`

Reply:
went ahead and implemented `handleInternalException`